web/src/message_actions_popover.ts

```diff
@@ -227,15 +228,17 @@ export function initialize(): void {
                 });
                 popover_menus.hide_current_popover_if_visible(instance);
             });
-
-            new ClipboardJS(the($popper.find(".copy_link"))).on("success", () => {
+            $popper.on("click", ".copy_link", () => {
+                const clipboardText = String($(".copy_link").data("clipboard-text"));
+                void clipboard_handler
+                    .copy_to_clipboard(clipboardText)
+                    .then(() => {
+                        popover_menus.hide_current_popover_if_visible(instance);
+                    })
+                    .catch((error: unknown) => {
+                        blueslip.error("Failed to copy to clipboard: ", {error: String(error)});
+                    });
                 show_copied_confirmation(the($(instance.reference).closest(".message_controls")));
```

Comment:
These handlers all seem essentially identical. I think you can probably use a shared function that accepts `instance`, `zulip_link_text`, and an option `success_callback` to handle the `show_copied_confirmation` call here, which probably wants to be in the `.then` logic here?